src/common/dsp/FMOscillator.cpp

```diff
@@ -78,38 +100,54 @@ void FMOscillator::process_block(float pitch, float drift, bool stereo, bool FM,
 
 void FMOscillator::init_ctrltypes()
 {
-   oscdata->p[0].set_name("M1 Amount");
-   oscdata->p[0].set_type(ct_percent);
-   oscdata->p[1].set_name("M1 Ratio");
-   oscdata->p[1].set_type(ct_fmratio);
-
-   oscdata->p[2].set_name("M2 Amount");
-   oscdata->p[2].set_type(ct_percent);
-   oscdata->p[3].set_name("M2 Ratio");
-   oscdata->p[3].set_type(ct_fmratio);
-
-   oscdata->p[4].set_name("M3 Amount");
-   oscdata->p[4].set_type(ct_percent);
-   oscdata->p[5].set_name("M3 Frequency");
-   oscdata->p[5].set_type(ct_freq_audible);
-
-   oscdata->p[6].set_name("Feedback");
-   oscdata->p[6].set_type(ct_osc_feedback);
+   oscdata->p[fm_m1amt].set_name("M1 Amount");
+   oscdata->p[fm_m1amt].set_type(ct_percent);
+   oscdata->p[fm_m1ratio].set_name("M1 Ratio");
+   oscdata->p[fm_m1ratio].set_type(ct_fmratio);
+
+   oscdata->p[fm_m2amt].set_name("M2 Amount");
+   oscdata->p[fm_m2amt].set_type(ct_percent);
+   oscdata->p[fm_m2ratio].set_name("M2 Ratio");
+   oscdata->p[fm_m2ratio].set_type(ct_fmratio);
+
+   oscdata->p[fm_m3amt].set_name("M3 Amount");
+   oscdata->p[fm_m3amt].set_type(ct_percent);
+   oscdata->p[fm_m3freq].set_name("M3 Frequency");
+   oscdata->p[fm_m3freq].set_type(ct_freq_audible);
+
+   oscdata->p[fm_fb].set_name("Feedback");
+   oscdata->p[fm_fb].set_type(ct_osc_feedback_negative);
 }
 void FMOscillator::init_default_values()
 {
-   oscdata->p[0].val.f = 0.0f;
-   oscdata->p[1].val.f = 1.0f;
-   oscdata->p[2].val.f = 0.0f;
-   oscdata->p[3].val.f = 1.0f;
-   oscdata->p[4].val.f = 0.0f;
-   oscdata->p[5].val.f = 0.0f;
-   oscdata->p[6].val.f = 0.0f;
+   oscdata->p[fm_m1amt].val.f = 0.0f;
+   oscdata->p[fm_m1ratio].val.f = 1.0f;
+   oscdata->p[fm_m2amt].val.f = 0.0f;
+   oscdata->p[fm_m2ratio].val.f = 1.0f;
+   oscdata->p[fm_m3amt].val.f = 0.0f;
+   oscdata->p[fm_m3freq].val.f = 0.0f;
+   oscdata->p[fm_fb].val.f = 0.0f;
+}
+
+void FMOscillator::handleStreamingMismatches(int streamingRevision, int currentSynthStreamingRevision)
+{
+   oscdata->p[fm_fb].set_type(ct_osc_feedback);
```

Comment:
You need to only do this if streamingReivsion = 12 not in all cases. (New patches shouldn't do this reset).